File: backend/app/schemas/alert.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


# ── Alert Rules ──────────────────────────────────────────────

class AlertRuleCreate(BaseModel):
    name: str
    description: str | None = None
    severity: str  # info | warning | critical
    conditions: dict = {}
    channels: list[str] = []  # ["email", "slack"]
    is_active: bool = True


class AlertRuleUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    severity: str | None = None
    conditions: dict | None = None
    channels: list[str] | None = None
    is_active: bool | None = None


class AlertRuleResponse(BaseModel):
    id: UUID
    name: str
    description: str | None
    severity: str
    conditions: dict
    channels: list[str]
    is_active: bool
    created_by: UUID | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# ── Alert Events ─────────────────────────────────────────────

class AlertEventResponse(BaseModel):
    id: UUID
    alert_rule_id: UUID
    case_id: UUID | None
    severity: str
    channel: str
    delivery_status: str
    trigger_info: dict
    delivered_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class AlertEventList(BaseModel):
    items: list[AlertEventResponse]
    total: int
    page: int
    size: int

File: frontend/src/types/alert.ts
// ── Alert Rules ──────────────────────────────────────────────

export type AlertChannel = 'email' | 'slack'
export type AlertDeliveryStatus = 'pending' | 'delivered' | 'failed'

export interface AlertRule {
  id: string
  name: string
  description: string | null
  severity: string
  conditions: Record<string, unknown>
  channels: string[]
  is_active: boolean
  created_by: string | null
  created_at: string
  updated_at: string
}

export interface AlertRuleCreate {
  name: string
  description?: string
  severity: string
  conditions?: Record<string, unknown>
  channels?: string[]
  is_active?: boolean
}

export interface AlertRuleUpdate {
  name?: string
  description?: string
  severity?: string
  conditions?: Record<string, unknown>
  channels?: string[]
  is_active?: boolean
}

// ── Alert Events ─────────────────────────────────────────────

export interface AlertEvent {
  id: string
  alert_rule_id: string
  case_id: string | null
  severity: string
  channel: string
  delivery_status: AlertDeliveryStatus
  trigger_info: Record<string, unknown>
  delivered_at: string | null
  created_at: string
}

export interface AlertEventList {
  items: AlertEvent[]
  total: number
  page: number
  size: number
}
